src/ui/MessageInput/index.jsx

```diff
@@ -1,103 +1,330 @@
 import React, {
-  useState,
-  useRef,
-  useEffect,
-  useContext,
+  useState, useRef, useEffect, useCallback, useContext, useMemo,
```

Comment:
nit: move to perline if more than 3 imports -> I thought the lint rule was there for this